### ISSUE TITLE
Laser Weapon Destroys Self When moving

### ISSUE THREAD

**Issue by Claakin** [tsunamayo/Starship-EVO#3001]
### Short Description of the issue:
firing the laser weapon while moving causes the weapon to destroy its barrel, or the entire weapon. 

...

### Steps to reproduce the issue:

Create large laser, attach barrel, fly forward, fire weapons.

...

### Environment
- Starship EVO version: 20w37a
- Graphics Card: 1080 Extreme Gaming

### Additional information:

I have uploaded photos and blueprint files.

It appears that the barrel does not change the poition at which the laser is generated, and the laser is still able to make contact with it's own gun/barrel. I say this because it only destroys the barrel on larger weapons, but when firing multiple weapons maybe lag causes the round to fire later, making it appear behind the weapon.  Maybe the speed of the round is not fast enough, maybe speeding up the shot would allow it to pass the barrel. ultimately, if it were generated a couple units in front of the barrel, somethig small and unnoticeable it may prevent this.

Another option is to disable friendly fire for the gun and barrel.
![gunfire moving fail 1](https://user-images.githubusercontent.com/69943614/94518443-8b407180-01ef-11eb-9bbb-f994a5b985a5.jpg)
![gunfire moving fail 2](https://user-images.githubusercontent.com/69943614/94518444-8bd90800-01ef-11eb-889e-55637aa48f77.jpg)
![gunfire moving fail 3](https://user-images.githubusercontent.com/69943614/94518445-8c719e80-01ef-11eb-83f9-e8d25a5c219d.jpg)
![gunfire moving fail 4](https://user-images.githubusercontent.com/69943614/94518446-8c719e80-01ef-11eb-8f1c-eeea25d62c9e.jpg)
[Draco.zip](https://github.com/tsunamayo/Starship-EVO/files/5296762/Draco.zip)

_(Screenshots, Videos, Error logs, Related issues)_

Please also include the `output_log.txt` file. You can find your log here: `%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\`

If you issue concerns a particular design please include any blueprint that could help to reproduce the issue.
Blueprint folder is over at:
`%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\Save_Data\Blueprints\`

**Comment by Claakin** [comment]
Here is the output log!

[output_log.txt](https://github.com/tsunamayo/Starship-EVO/files/5296769/output_log.txt)

**Comment by matejkar** [comment]
are those turrets or fixed weapons? If they are fixed try to max out the barrels and also u have gatlings there meaning more spread.
Stuff like this happened to me in smaller builds when i try to put guns on spots where the ship is almost in a line of fire

**Comment by Claakin** [comment]
I'll try it without the gatling but th point is rate of fire, so I'm going to be losing a lot of potential damage. I mean we're already woring with unreal physics of bullets following a cursor fro ma stationary weapon which is fine, It helps, but the gun should not destroy its self no matter what you use.

**Comment by tsunamayo** [comment]
Yes the laser is supposed to be spawned at the muzzle position of the barrel, with a small offset, with the speed of the ship added. But there might be still something bugged as you pointed out.
I keep you posted

**Comment by Claakin** [comment]
Thank you, no problem! Keep up the good work! Can't wait for an update, I
know you're working hard!

**Comment by Rifleman75** [comment]
I'm having this happen pretty consistently at speeds over 250m/s.
I moved the lasers away from the ship and they then hit themselves and destroy either the barrels or laser module.
Gatling barrels seem the worst.

**Comment by tsunamayo** [comment]
@Rifleman75 @Claakin  Could you send me your blueprint? Thanks
